animation_nodes/nodes/mesh/set_vertex_weights.py

```diff
@@ -0,0 +1,131 @@
+import bpy
+from bpy.props import *
+from ... events import propertyChanged
+from ... data_structures import VirtualDoubleList
+from ... base_types import AnimationNode, VectorizedSocket
+
+modeItems = [
+    ("ALL", "All", "Set weight of every vertex", "NONE", 0),
+    ("INDEX", "Index", "Set weight of a specific vertex", "NONE", 1)
+]
+
+groupIdentifierTypeItems = [
+    ("INDEX", "Index", "Get vertex group based on the index", "NONE", 0),
+    ("NAME", "Name", "Get vertex group based on the name", "NONE", 1)
+]
+
+groupNotFoundMessage = "group not found"
+noMeshMessage = "no mesh object"
+editMeshMessage = "object is not in object mode"
+
+class VertexWeights(bpy.types.Node, AnimationNode):
+    bl_idname = "an_SetVertexWeights"
+    bl_label = "Set Vertex Weights"
+    errorHandlingType = "EXCEPTION"
+
+    mode: EnumProperty(name = "Mode", default = "ALL",
+        items = modeItems, update = AnimationNode.refresh)
+
+    groupIdentifierType: EnumProperty(name = "Group Identifier Type", default = "INDEX",
+        items = groupIdentifierTypeItems, update = AnimationNode.refresh)
+
+    useIndexList: VectorizedSocket.newProperty()
+    useWeightsList: VectorizedSocket.newProperty()
+ 
+    def create(self):
+        self.newInput("Object", "Object", "object", defaultDrawType = "PROPERTY_ONLY")
+
+        if self.groupIdentifierType == "INDEX":
+            self.newInput("Integer", "Group Index", "groupIndex")
+        elif self.groupIdentifierType == "NAME":
+            self.newInput("Text", "Name", "groupName")
+
+        if self.mode == "INDEX":
+            self.newInput(VectorizedSocket("Integer", "useIndexList",
+                ("Index", "index"), ("Indices", "indices")))
+        self.newInput(VectorizedSocket("Float", "useWeightsList",
+                ("Weight", "weight"), ("Weights", "weights")))
+        
+        self.newOutput("Object", "Object", "object")
+
+    def draw(self, layout):
+        layout.prop(self, "mode", text = "")
+
+    def drawAdvanced(self, layout):
+        layout.prop(self, "groupIdentifierType", text = "Type")
+
+    def getExecutionFunctionName(self):
+        if self.mode == "INDEX":
+            if self.useIndexList:
+                if self.useWeightsList:
+                    return "execute_IndicesList"
+                else:
+                    return "execute_IndicesSingle"
+            else:
+                return "execute_Index"
```

Comment:
What will happen if `useIndexList` is false and `useWeightsList` is true? The `execute_Index` function will be called and the `weight` argument would be a list, not a float.

Reply:
Then, it gives the error:
![Screenshot from 2019-11-10 10-57-30](https://user-images.githubusercontent.com/30294746/68539388-f5b6ac00-03a8-11ea-8a41-2db6488e1bf7.png)

Yes, you are right, `weight` argument would be a list. So I think, we add another function which handles this:
```
def execute_Index_WeightsList(self, object, identifier, index, weights):
        if object is None: return
        vertexGroup = self.getVertexGroup(object, identifier)

        vertexGroup.add([index], weights[0], "REPLACE")
        object.data.update()    
        return object
```
or do you have any other idea?

Reply:
I believe we can solve this by adding the `useIndexList` to the list of vectorized socket properties like so:

```
self.newInput(VectorizedSocket("Float", ["useWeightsList", "useIndexList"],
    ("Weight", "weight"), ("Weights", "weights")))
```
I think this will break the ALL case though, so we might need to branch again. But I haven't tried that, can you try that?

Reply:
Okay. I'll try and let you know.

Reply:
It worked without any error for all the modes (Index, Indices, and All).